buildchain/buildchain/image.py

```diff
@@ -87,7 +87,7 @@ def show() -> str:
 
 
 NGINX_IMAGE_VERSION : str = '1.15.8'
-
+NODE_IMAGE_VERSION : str = '10.16.0'
```

Comment:
```
-update file_dep
-change the pull policy deployment.yaml
-remove useless html element
```

Those 3 changes are totally unrelated (unless I miss something) and should have dedicated commits.

A commit is "atomic" and has one goal: what if I want to revert/cherry-pick the removal of the useless HTML element without losing/picking the `file_dep` update?

Reply:
Two first changes are part of the build configuration. Ok the last one is a missing error but I will not create another for 2 characters removal.

Reply:
the deployment.yaml is not related to the buildchain, but to the deployment so they don't belong to the same commit.

Having a dedicated commit for this one would allow you to explains why you added ` imagePullPolicy: Always` in the commit message (useful when someone will use `git log` or `git blame`).

And you should create a dedicated commit, even for your two characters removal since it has absolutely no relation with the rest (a little commit "fix typo" cost nothing and keep a usable and useful git history, which save a lot more time in the future than splitting the commit).

Reply:
Ok for the last one. but for the two first, I do not agree. It depend on how we define the "atomicity" of a commit.